Source/Manager.swift

```diff
@@ -107,7 +119,34 @@ public class Manager: NSURLProtocol {
         
         name = pathComponents.joinWithSeparator(separator)
         name = separator + name!
+
+        let query = queryDictionaryFromURL(url)
         
-        return (name, ids)
+        return (name, ids, query)
+    }
+
+    // inspired by https://gist.github.com/freerunnering/1215df277d750af71887
+    /// Return a dictionary with arrays populated with the values of query parameters. URLs allow keys to be declared multiple times, hence the array container.
+    static func queryDictionaryFromURL(url: NSURL) -> QueryArtifacts? {
+        guard let query = url.query else { return nil }
+
+        var queryDict = [String: [String]]()
```

Comment:
I _think_ you can use the typealias name as an initializer: `var queryDict = QueryArtifacts()`